docs/DesignDocs/SGX_Endorsements_V2.md

```diff
@@ -0,0 +1,222 @@
+
+##### tcbDate:
+	type: string
+	format: date-time
+	description: >- Representation of date and time when the TCB level was certified not to be vulnerable to any issues described in SAs that were published on or prior to this date. The time shall be in UTC and the encoding shall be compliant to ISO 8601 standard (YYYY-MM-DDThh:mm:ssZ).
+
+##### tcbStatus:
+	type: string
+	description: TCB level status
+
+##### advisoryIDs:
+	type: array
+	description: >- Array of Advisory IDs describing vulnerabilities that this TCB level is vulnerable to. Note: The value can be different for different FMSPCs. This field is optional. It will be present only if the list of Advisory IDs is not empty.
+	items:
+        type: string
+
+For more information on the new fields please see [API Version 2](https://api.portal.trustedservices.intel.com/documentation#pcs-certificate-v2).
+
+
+## Changes
+
+### Update current structs with new additional fields and create new type definitions.
+
+```C
+typedef enum _oe_tcb_level_status
+{
+    // Existing fields
+    OE_TCB_LEVEL_STATUS_UNKNOWN,
+    OE_TCB_LEVEL_STATUS_REVOKED,
+    OE_TCB_LEVEL_STATUS_OUT_OF_DATE,
+    OE_TCB_LEVEL_STATUS_CONFIGURATION_NEEDED,
+    OE_TCB_LEVEL_STATUS_UP_TO_DATE,
+
+    // New field for new value "OutOfDateConfigurationNeeded"
+    OE_TCB_LEVEL_STATUS_OUTOFDATE_CONFIGURATION_NEEDED,
+
+    __OE_TCB_LEVEL_MAX = OE_ENUM_MAX,
+
+} oe_tcb_level_status_t;
+
+// Existing TCB Level struct for TCB Info.  Will rename this
+// struct to oe_tcb_tcb_level given that
+// QE identity info also has its TCB level data field.
+typedef struct _oe_tcb_level
+{
+    // Existing fields
+    uint8_t sgx_tcb_comp_svn[16];
+    uint16_t pce_svn;
+    oe_tcb_level_status_t tcb_status;
+
+    // New fields
+    oe_datetime_t tcb_date;
+    uint8_t* advisory_ids_json;     // ["INTEL-SA-00079", "INTEL-SA-00076"]
+
+} oe_tcb_level_t;
+
+typedef struct _oe_parsed_tcb_info
+{
+    // Existing fields
+    uint32_t version;
+    oe_datetime_t issue_date;
+    oe_datetime_t next_update;
+    uint8_t fmspc[6];
+    uint8_t pceid[2];
+    uint8_t signature[64];
+    const uint8_t* tcb_info_start;
+    size_t tcb_info_size;
+
+    // New fields
+    uint32_t tcb_type;
+    uint32_t tcb_evaluation_data_number;
+
+} oe_parsed_tcb_info_t;
+
+/*!
+ * New enum for QE id field
+ */
+typedef enum _oe_qe_identity_id
+{
+    OE_IDENTITY_ID_QE,
+    QE_IDENTITY_ID_QVE
+} oe_qe_identity_id;
+
+/*!
+ * New TCB level for QE identity info.
+ */
+typedef struct _oe_qe_tcb_level
+{
+    uint32_t isvsvn;
+    oe_tcb_level_status_t tcb_status;
+    oe_datetime_t tcb_date;
+    uint8_t* advisory_ids_json;     // ["INTEL-SA-00079", "INTEL-SA-00076"]
+
```

Comment:
what are ["INTEL-SA-00079", "INTEL-SA-00076"]?

Reply:
These are the values from the new advisoryIDs field.  This field is an array of strings that describe the vulnerabilities that the corresponding TCB level is vulnerable to.  The value can be different for each FMSPCs, and it is an optional field.

Reply:
I see. Could add those information into the doc?

Reply:
There is already a section "New field definitions" in the doc that has a description of this new field.